### PR TITLE
cache and typo fix

### PR DESCRIPTION
**Pull request by Bowrna**
Typo fix in Breeze command line instruction
Cache - use the airflow cache 

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Review by uranusjr — APPROVED** [review]

**Comment by potiuk** [comment]
We can't use cache from functools as it is Python 3.9+:

https://docs.python.org/3/library/functools.html#functools.cache

> New in version 3.9.

**Comment by Bowrna** [comment]
> We can't use cache from functools as it is Python 3.9+:
> 
> https://docs.python.org/3/library/functools.html#functools.cache
> 
> > New in version 3.9.

@potiuk this code in airflow's functool picks up the cache based on python version  airflow/compat/functools.py

https://github.com/apache/airflow/blob/0ae0f7e2448e05917e51e29b854ad60463378fbe/airflow/compat/functools.py#L18-L33

**Comment by Bowrna** [comment]
@potiuk static checks are failing in `sqlalchemy` part. I am not sure why it's failing and the above import of cache from `airflow/compat/functools.py` doesn't have dependency on `sqlalchemy`  package. Am I missing to understand any part?

**Comment by potiuk** [comment]
@potiuk this code in airflow's functool picks up the cache based on python version airflow/compat/functools.py

Yes. But we always run breeze with Python 3.7, AND breeze should not import "airflow". 

This is also the reason for sqlalchemy failing. 

Breeze "project" is separate from Airflow - and it should not import "airflow" - because airflow requires  far more dependencies. I even just added a pre-commit check for for that in #22880 - (limit-breeze-dependencies) that makes sure that `breeze.py` can imported and run with just `rich` and `click` imported. If you would like Breeze to depend on Airflow - it would mean MANY more dependencies, which we do not want to install for Breeze.

File: dev/breeze/src/airflow_breeze/utils/run_utils.py [hunk 1]
```diff
@@ -26,6 +26,7 @@
 from pathlib import Path
 from typing import Dict, List, Mapping, Optional
 
+from airflow.compat.functools import cache
```
Comment (potiuk) [inline dev/breeze/src/airflow_breeze/utils/run_utils.py]: This "import" is wrong. Breeze has no access to "airflow" package and installing "airflow" for Breeze is no-go due to number of dependencies.

Comment (Bowrna) [reply dev/breeze/src/airflow_breeze/utils/run_utils.py]: i see. i could understand this now. thanks @potiuk i will revert the changes

**Comment by Bowrna** [comment]
@potiuk @uranusjr I have reverted the cache related changes. That leaves this PR with small typo fix alone. But i got to learn why i shouldn't import from airflow package. :)

**Review by uranusjr — DISMISSED** [review]

**Comment by uranusjr** [comment]
Needs a rebase though, the doc was rewritten extensively.

**Comment by Bowrna** [comment]
@potiuk @uranusjr 

This is the error in static check. This error doesn't occur when i run pre-commit locally. But this occurs in CI. I am not sure how to fix it. 

```
Update output of breeze commands in BREEZE.rst...........................................Failed
- hook id: update-breeze-file
- exit code: 1

Traceback (most recent call last):
  File "./scripts/ci/pre_commit/pre_commit_breeze_cmd_line.py", line 81, in <module>
    print_help_for_all_commands()
  File "./scripts/ci/pre_commit/pre_commit_breeze_cmd_line.py", line 46, in print_help_for_all_commands
    check_call(["breeze", "--help"], env=env)
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/subprocess.py", line 358, in check_call
    retcode = call(*popenargs, **kwargs)
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/subprocess.py", line 339, in call
    with Popen(*popenargs, **kwargs) as p:
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/subprocess.py", line 800, in __init__
    restore_signals, start_new_session)
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/subprocess.py", line 1551, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'breeze': 'breeze'
```

**Comment by uranusjr** [comment]
This seems to be caused by #22880; I think it’s because breeze is not installed in the CI environment.

**Comment by potiuk** [comment]
Rebase. We have now completely different pre-commit in place of updating the help output.

Instead we generate nice .svg output (using rich + rich-click + rich built-in svg export capabilities) to update the images that are embedded in the BREZEE.rst so the old static check would not work.

**Comment by Bowrna** [comment]
Could you explain more about .svg @potiuk ? You could share any relevant docs or point to links to know more about this.

**Comment by Bowrna** [comment]
@potiuk I have rebased the code now and still issue persists. Am i missing to understand any part?

**Comment by potiuk** [comment]
Ah I see - "Basic checks only"!  @uranusjr was right - this CI step should have breeze installation same as the "static checks" - you can copy it from "Static Checks" in "ci.yml"

The problem in this case was an interesting one :) 

- you had a change that did not require building image (because only .rst file changed that was not part of the docs)
- therefore instead of "full static chcecks" only "basic ones" triggered
- And the "basic checks" did not have Breeze installed.

**Comment by potiuk** [comment]
I will also add a nicer failure in this check though. It should inform the user that in order to run this check, they need to have breeze installed in the error.

**Comment by potiuk** [comment]
Added issue for "nicer" message https://github.com/apache/airflow/issues/22908 but for now you can turn this one into PR fixing the problem by adding Breeze installation in the CI job., @edithturn might help if needed.

**Comment by Bowrna** [comment]
:) static basic check is updated with breeze and pipeline is success

**Review by potiuk — APPROVED** [review]

**Comment by edithturn** [comment]
@Bowrna @potiuk I am trying to understand what is happening, I think there is a conflict in this file https://github.com/apache/airflow/blob/main/BREEZE.rs , it was already merged. 

ksnip_20220412-110556

Let me know if this comment doesn't align with what you're already doing.
t

**Comment by potiuk** [comment]
Ah yeah. I merged it too fast :(

**Comment by potiuk** [comment]
Github simply "hidden" the change and I have not realized it's there!

**Comment by edithturn** [comment]
I was checking and it's just removing the **Head** and the bottom. I can do it : >>>>>>> 5c42776f29 (cache and typo fix)

**Comment by edithturn** [comment]
I think that should be noted in github-commits before pushing it

**Comment by edithturn** [comment]
Let me do it Jarek

**Comment by potiuk** [comment]
Bad me https://github.com/apache/airflow/pull/22953

**Comment by potiuk** [comment]
ah.. I was too fast :)

**Pull request merged**